testpmd/README.md

```diff
@@ -1,55 +1,130 @@
 
-# Stand-alone-testpmd
+# Testpmd with REST API
 
-Testpmd is commonly used in DPDK performance test.
-It makes sense to have the testpmd seperated from the common tool set container and have its own 
-container image. This allows us to quickly add new capabilities to the testpmd without 
-worrying the update impact to other tools in the common tool set container.
+Intel DPDK sample application testpmd is commonly used in DPDK performance test.
+
+In this repository, a wrapper REST API is added to the testpmd, so that the testpmd can be remotely started/stopped/queried.
 
 ## Prerequisites:
 + 2MB or 1GB huge pages
 + isolated CPU for better performance
 + in BIOS, enable VT (cpu virtualization technology)
-+ intel_iommu in kernel argument
++ iommu enabled
 + Example kargs: `default_hugepagesz=1G hugepagesz=1G hugepages=8 intel_iommu=on iommu=pt isolcpus=4-11`
++ two testpmd ports are located on the same numa node and pre-bound to the vfio-pci driver
 
-## Openshift integration demo
-
-[![Watch the video](https://img.youtube.com/vi/C5s9DZC3D6c/hqdefault.jpg)](https://youtu.be/C5s9DZC3D6c)
+To achieve higher traffic rate, the two testpmd ports should come from different NICs.
 
-## Podman run example
+## Build the container image
 
-### start testpmd server 
+To build the container image:
+```
+podman build -t testpmd:22.11.2 .
+```
 
-`podman run -it --rm --privileged -p 9000:9000 -v /sys:/sys -v /dev:/dev -v /lib/modules:/lib/modules --cpuset-cpus 5,7,9,11 docker.io/cscojianzhan/testpmd /root/testpmd-wrapper -pci 86:00:0 -pci 86:00:1`
+This image size is around 175M bytes.
 
-### control testpmd from a client
+The DPDK version can be changed by using `build-arg` build option, for example, so build 21.11.2,
```

Comment:
remove ", so": "for example build 21.11.2"

Reply:
ack